packages/near-fast-auth-signer/src/lib/controller.ts

```diff
@@ -205,6 +209,54 @@ class FastAuthController {
     return signedDelegate;
   }
 
+  async signTransaction({ receiverId, actions, signerId }:
+    { receiverId: string; actions: Action[]; signerId: string }):
+   Promise<[Uint8Array, SignedTransaction]> {
+    this.assertValidSigner(signerId);
+    const account = new Account(this.connection, this.accountId);
+
+    const accessKeyInfo = await account.findAccessKey(receiverId, actions);
+    if (!accessKeyInfo) {
+      throw new TypedError(
+        `Can not sign transactions for account ${this.accountId} on network ${this.connection.networkId}, no matching key pair exists for this account`,
+        'KeyNotFound'
+      );
+    }
+    const { accessKey } = accessKeyInfo;
+
+    const block = await this.connection.provider.block({
+      finality: 'final',
+    });
+    const blockHash = block.header.hash;
+
+    const nonce = accessKey.nonce.add(new BN(1));
+
+    if (await isPassKeyAvailable()) {
+      return signTransaction(
+        receiverId,
+        nonce,
+        actions,
+        baseDecode(blockHash),
+        this.connection.signer,
+        this.accountId,
+        this.connection.networkId
+      );
+    }
+
+    const oidcToken = await firebaseAuth.currentUser.getIdToken();
+    const localKey = await this.getKey(`oidc_keypair_${oidcToken}`) || await this.getLocalStoreKey(`oidc_keypair_${oidcToken}`);
```

Comment:
This is the FRP key so it cannot be used to sign anything. If we want to implement fallback to signing with the oidc token then we need to make a request to the mpc service. IMO we can consider treating sign routes as iframe only and if we do then there is no need for fallback to oidc because iframes will not have access to that either way

Reply:
Agree, I will remove the fallback code.